docs/programming_guide/experiment_tracking/experiment_tracking_apis.rst

```diff
@@ -0,0 +1,211 @@
+Here we will discuss the following topics:
+
+- Logging metrics with MLflow, TensorBoard, or Weights & Biases
+- Streaming metrics to the FL server
+- Streaming to FL clients
+
+Logging metrics with MLflow, TensorBoard, or Weights & Biases
+=============================================================
+
+Integrate MLflow logging to efficiently stream metrics to the MLflow server with just three lines of code:
+
+.. code-block:: python
+
+  from nvflare.client.tracking import MLflowWriter
+
+  mlflow = MLflowWriter()
+
+  mlflow.log_metric("loss", running_loss / 2000, global_step)
+
+In this setup, we use ``MLflowWriter`` instead of using the MLflow API directly.
+This abstraction is important, as it enables users to flexibly redirect your logging metrics to any destination, which we discuss in more detail later.
+
+The use of MLflow, TensorBoard, or Weights & Biases syntax will all work to stream the collected metrics to any supported experiment tracking system.
+Choosing to use TBWriter, MLflowWriter, or WandBWriter is user preference based on your existing code and requirements.
+
+- ``MLflowWriter`` uses the Mlflow API operation syntax ``log_metric()``
+- ``TBWriter`` uses the TensorBoard SummaryWriter operation ``add_scalar()``
+- ``WandBWriter`` uses the Weights & Biases API operation ``log()``
+
+Here are the APIs:
+
+.. code-block:: python
+
+  class TBWriter(LogWriter):
+    def add_scalar(self, tag: str, scalar: float, global_step: Optional[int] = None, **kwargs):
+    def add_scalars(self, tag: str, scalars: dict, global_step: Optional[int] = None, **kwargs):
+
+
+  class WandBWriter(LogWriter):
+    def log(self, metrics: Dict[str, float], step: Optional[int] = None):
+
+
+  class MLflowWriter(LogWriter):
+      def log_param(self, key: str, value: any) -> None:
+      def log_params(self, values: dict) -> None:
+      def log_metric(self, key: str, value: float, step: Optional[int] = None) -> None:
+      def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+      def log_text(self, text: str, artifact_file_path: str) -> None:
+      def set_tag(self, key: str, tag: any) -> None:
+      def set_tags(self, tags: dict) -> None:
+
+
+After you've modified the training code, you can use the NVFlare's job configuration to configure the system to stream the logs appropriately.
+
+Streaming metrics to FL server
+==============================
+
+All metric key values are captured as events, with the flexibility to stream them to the most suitable destinations.
+Let's add the ``ConvertToFedEvent`` to convert these metrics events to federated events so they will be sent to the server.
+
+Add this component to config_fed_client.json:
+
+.. code-block:: yaml
+
+  {
+    "id": "event_to_fed",
+    "name": "ConvertToFedEvent",
+    "args": {"events_to_convert": ["analytix_log_stats"], "fed_event_prefix": "fed."}
+  }
```

Comment:
When is this used? InProcessExecutor? Do we need to specify that then?

Reply:
you're right, note that "MetricRelay" converts to fed_event automatically, so we don't need this component if MetricRelay is used.

This as of now can be here, let's try to make users experience better regarding these metrics streaming configuration later.